lsp/server/src/validateDocument.ts

```diff
@@ -45,10 +45,5 @@ export async function validateDocument(
         results = results.concat(diagnostics);
     }
 
-    // Set the source for diagnostic source.
-    results.forEach((diagnostic) => {
-        diagnostic.source = extensionName;
```

Comment:
Curious, why did you get rid of this? I wasn't a fan of this post processing anyway so removing it is a welcome change but where is the source set now?

Reply:
The vscode has its own brain where and how to render the diagnostic attributes.  I like to render the diagnostic.source "Salesforce Mobile Extension in ui so to tell developer where this diagnostic is from. When we try to render a link in diagnostic pop up like what Kevin wants.  the source is rendered between the msg and cde/codeDescription like below. Totally messed up the effect we want.  Removing the source makes the popup close to what we want. 
<img width="779" alt="image" src="https://github.com/user-attachments/assets/9cce0852-ac17-435e-838b-fe28db5b87f6">

Source will not be set if we will render the link. 
Kevin is thinking of what will be finally word for the rules, if there's no link needed.  we can put the source back.

Reply:
To make myself clear, I wasn't the fan of post processing, so removing is fine I think. But do what you think is the best thing to do.